pyuvsim/uvsim.py

```diff
@@ -444,16 +444,17 @@ def run_uvdata_uvsim(input_uv, beam_list, beam_dict=None, catalog=None, quiet=Fa
     if Nsky_parts > Nsrcs:
         raise ValueError("Insufficient memory for simulation.")
 
-    Ntasks_tot = Ntimes * Nbls * Nfreqs * Nsky_parts
-
     local_task_iter = uvdata_to_task_iter(
         task_inds, input_uv, catalog.subselect(src_inds),
         beam_list, beam_dict, Nsky_parts=Nsky_parts
     )
 
-    Ntasks_tot = comm.reduce(Ntasks_tot, op=mpi.MPI.MAX, root=0)
+    Ntasks_tot = Ntasks_local * Nsky_parts
+    # Sum all the tasks across each node
+    Nsky_parts = comm.reduce(Nsky_parts, op=mpi.MPI.MAX, root=0)
+    Ntasks_tot = comm.reduce(Ntasks_tot, op=mpi.MPI.SUM, root=0)
     if rank == 0 and not quiet:
-        print(f"Nsky parts: {Nsky_parts}", flush=True)
+        print(f"Max Nsky parts: {Nsky_parts}", flush=True)
```

Comment:
Looking at this again, this might be a good opportunity to make the printouts a little more informative. Maybe the following?
```
ntot_str = "Ntimes x Nfreqs x Nbls"
if Nsky_parts > 1:
    print(f"The source list was split into Nsky_parts <= {Nsky_parts} chunks on some or all nodes due to memory limits.")
   tot_str += " x Nsky_parts"
print(f"Total tasks ({tot_str}): {Ntasks_tot}")
```

Reply:
Yeah this sounds good. I am not sure I want to add the Ntimes x Nfreqs x Nsky_parts = Ntasks_tot though because that is not necessarily true.

Reply:
Ah okay now it is complaining about coverage with that if statement. I suppose it could be possible to cook up a test to force nsky_parts to be more than one.

Reply:
Hm... there is a test in `test_mpi.py` that fakes environmental variables so the test thinks the memory is low, and it will split the sources up. I'm not sure why that isn't hitting this line.

Reply:
I cannot find the one you mentioned, but found `test_source_splitting`  in test_uvsim.py that does set slurm environment variables. but it does not call `run_uvdata_uvsim` but just checks the logic of it.

Reply:
Ah, that must be the one I'm thinking of. It seems a bit silly to write a whole new test for this, but if you're up for it it shouldn't be too painful to manually set that environmental variable for a quick `test_mpi.py` test.